### PR TITLE
Feature/docs public deploy #1083

### PR DESCRIPTION
**Pull request by abcoathup**
Port of forum guide (https://forum.openzeppelin.com/t/deploy-to-a-public-testnet-using-zeppelinos-2-4/941) to the documentation.

Pull Request for Issue: https://github.com/openZeppelin/openzeppelin-sdk/issues/1083

Let me know of any changes required to match the documentation style.

### CODE REVIEW

File: packages/docs/modules/ROOT/pages/public_deploy.adoc [hunk 1]
```diff
@@ -0,0 +1,185 @@
+[[deploy-your-contracts-to-a-public-network]]
+= Deploy your contracts to a public network
+
+This tutorial link:first[builds on the first tutorial] where we created a new OpenZeppelin SDK project, and created a simple `Counter` contract in a local development network. We will now see how to deploy this contract to a public network.  
+
+We will use the Ropsten public test network for this tutorial.  You could also use other public test networks such as Rinkeby, Kovan or Goerli.
```
Comment (spalladino) [inline packages/docs/modules/ROOT/pages/public_deploy.adoc]: Let's add a note block here briefly explaining the differences between the testnets. At least, let's point out that ropsten is PoW while the others are PoA (and what that implies in terms of block times), and that Rinkeby and Kovan only work with Geth and Parity respectively.

File: packages/docs/modules/ROOT/pages/public_deploy.adoc [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+We will use https://github.com/motdotla/dotenv[dotenv] to store your Infura Project ID and your development mnemonic.
+Run the following to install as a development dependency:
+
+[source,console]
+----
+$ npm install --save-dev dotenv
+----
+
+[[configure-network-js]]
+== Configure network.js
```
Comment (spalladino) [inline packages/docs/modules/ROOT/pages/public_deploy.adoc]: The file name is `networks.js` (plural), not `network.js`. Make sure to change this across the doc!

Comment (abcoathup) [reply packages/docs/modules/ROOT/pages/public_deploy.adoc]: Oops, sorry about that.  Corrected.

File: packages/docs/modules/ROOT/pages/public_deploy.adoc (hunk 1 above)
Comment (abcoathup) [reply packages/docs/modules/ROOT/pages/public_deploy.adoc]: Added, though not sure how to add new lines in a Note.

Comment (abcoathup) [reply packages/docs/modules/ROOT/pages/public_deploy.adoc]: Added

File: packages/docs/modules/ROOT/pages/public_deploy.adoc (hunk 2 above)
Comment (abcoathup) [reply packages/docs/modules/ROOT/pages/public_deploy.adoc]: Oops, resolved.

**Comment by abcoathup** [comment]
Updated, reordered steps as per your recommendation.  Combined sentences into single paragraphs and made some minor edits.

**Pull request merged**